### ISSUE TITLE
not-callable: Different behavior using f-string vs bin-op string

### ISSUE THREAD

**Issue by moylop260** [PyCQA/pylint#10422]
### Bug description

Using the following code:

```python
class MyClass():
    meth_name = fields.Char(default="test")

    def _call_test(self):
        pass

    def _call_provider(self):
        method = getattr(self, f"_call_{self.meth_name}", None)  # not-callable raised
        method()
        method2 = getattr(self, "_call_%s" % self.meth_name, None)  # not-callable not raised
        method2()
```

Call the following pylint command:

`pylint -d all -e not-callable my_test.py`

Notice the following output:

`my_test.py:9:8: (not-callable) method is not callable`

Notice we are using the same attribute class for both cases, for f-String and BinOp

But only the f-string raises the check

The expected behaviour does not raise the check error since the attribute class `meth_name` can not be correctly inferred in this point

### Configuration

```ini

```

### Command used

```shell
pylint -d all -e not-callable --report n --score n my_test.py
```

### Pylint output

```txt
my_test.py
************* Module my_test
my_test.py:9:8: E1102: method is not callable (not-callable)
```

### Expected behavior

Same behaviour using F-string and BinOp

In this case, BinOp is correct not raising the check

f-string should not raise this check too

### Pylint version

```shell
Tested with different environments:

(latest github main branch)

pylint 4.0.0-dev0
astroid 3.3.10
Python 3.13.3

(latest stable pypi)

pylint 3.3.7
astroid 3.3.10
Python 3.13.3 (main, Apr  8 2025, 13:54:08) [Clang 17.0.0 (clang-1700.0.13.3)]

Different python versions

pylint 3.3.7
astroid 3.3.10
Python 3.12.10 (main, Apr  8 2025, 11:35:47) [Clang 17.0.0 (clang-1700.0.13.3)]

pylint 3.3.7
astroid 3.3.10
Python 3.10.17 (main, Apr  8 2025, 12:10:59) [Clang 16.0.0 (clang-1600.0.26.6)]

```

### OS / Environment

MacOSX
Ubuntu

### Additional dependencies

```python

```
